secretmanager/src/main/java/secretmanager/regionalsamples/UpdateRegionalSecret.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package secretmanager.regionalsamples;
+
+// [START secretmanager_update_regional_secret]
+import com.google.cloud.secretmanager.v1.Secret;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1.SecretManagerServiceSettings;
+import com.google.cloud.secretmanager.v1.SecretName;
+import com.google.protobuf.FieldMask;
+import com.google.protobuf.util.FieldMaskUtil;
+import java.io.IOException;
+
+public class UpdateRegionalSecret {
+
+  public static void updateRegionalSecret() throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "your-project-id";
+    String locationId = "your-location-id";
+    String secretId = "your-secret-id";
+    updateRegionalSecret(projectId, locationId, secretId);
+  }
+
+  // Update an existing secret.
+  public static void updateRegionalSecret(
+      String projectId, String locationId, String secretId) 
+      throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    
+    // Endpoint to call the regional secret manager sever
+    String apiEndpoint = String.format("secretmanager.%s.rep.googleapis.com:443", locationId);
+    SecretManagerServiceSettings secretManagerServiceSettings =
+        SecretManagerServiceSettings.newBuilder().setEndpoint(apiEndpoint).build();
+    try (SecretManagerServiceClient client = 
+        SecretManagerServiceClient.create(secretManagerServiceSettings)) {
+      // Build the name.
+      SecretName secretName = 
+          SecretName.ofProjectLocationSecretName(projectId, locationId, secretId);
+
+      // Build the updated secret.
+      Secret secret =
+          Secret.newBuilder()
+              .setName(secretName.toString())
+              .putLabels("secretmanager", "rocks")
+              .build();
+
+      // Build the field mask.
+      FieldMask fieldMask = FieldMaskUtil.fromString("labels");
```

Comment:
I have not tried but existing samples use the same way, eg. https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/secretmanager/src/main/java/secretmanager/UpdateSecret.java#L53

It might create confusion to some users if we do the same thing via two different approaches.